### DISCUSSION TITLE
Jump to a class in API reference

### DISCUSSION THREAD

**Discussion by graealex** [EWSoftware/SHFB#978] [Q&A] [answered]
My application exposes a lot of classes via TreeViews and PropertyGrid to the user. When hitting F1/Help, I want to redirect to the API reference documentation in a CHM file for that class. My questions are:

1. Is there a good way to directly launch a topic from API reference? Currently I use: `System.Windows.Forms.Help.ShowHelp(null, ProvideCHM(), System.Windows.Forms.HelpNavigator.KeywordIndex, $"{typeName} class, {typeName} Class");` but it's hit and miss, even if the keyword with the exact name exists. Or is there a better way?
2. How stable is the GUID topic generation for API reference? Could I just add an attribute to each class that references the exact topic GUID and be sure that it doesn't change?

**Comment by EWSoftware** [comment, marked as answer]
I used to have a help 1 file for SHFB and used the topic GUIDs when launching the viewer adding on an optional anchor for the section ID if necessary.

``` C#
Help.ShowHelp(form, path, HelpNavigator.Topic, "html/" + topic + ".htm" + anchor)
```

The GUIDs are actually MD5 hashes of the topic IDs in GUID form so they are constant.

**Reply by graealex** [reply]
Then I shall check SHFB source code for the MD5 hashing function, I probably should be able to create the MD5 hashes from the exact namespace and class name. I'd rather avoid having to decorate all classes with redundant information. Thank you!

**Comment by EWSoftware** [comment]
You may have found it but if not, it is [here](https://github.com/EWSoftware/SHFB/blob/master/SHFB/Source/SandcastleBuilderUtils/ApiTopicNamer.cs#L122)

**Reply by graealex** [reply]
Yes, I found it, and I decided to change the NamingMethod to MemberName, which makes it a lot easier to find the exact topic for a particular class, without hashing anything.
